src/controllers/user-controller.js

```diff
@@ -148,8 +211,63 @@ router.post("/login", async (req, res) => {
  * @return {object} 403 - Prohibitted
  * @return {object} 500 - Internal server error
  */
-router.post("/", async (req, res) => {
+router.post("/register", async (req, res) => {
     try {
+        const email = req.body.email;
+        const password = await bcrypt.hash(req.body.password, 10);
+        const userName = req.body.userName;
+        const userType = req.body.userType;
+
+        if (!email || !password || !userName || !userType) {
+            // Translation: Registeration requires: email, password, name, and account type.
+            return res.status(400).json({ error: "회원가입을 위해서는 다음 정보가 필요합니다: 이메일, 암호, 이름, 계정종류" });
```

Comment:
## Improvement
As I am not using name as part of the sign up process, the error message should indicate that user name is required instead of name.

Reply:
Modification has been made for this improvement.